dids/src/main/kotlin/web5/sdk/dids/methods/dht/DidDht.kt

```diff
@@ -45,393 +79,500 @@ public class DidDhtConfiguration internal constructor(
 public class CreateDidDhtOptions(
   public val verificationMethods: Iterable<Pair<JWK, Array<PublicKeyPurpose>>>? = null,
   public val services: Iterable<Service>? = null,
-  // TODO(gabe): flip this to true when publishing is implemented
-  public val publish: Boolean? = false,
+  public val publish: Boolean = true,
```

Comment:
I mirrored the go and ts APIs here, I don't see the problem with an optional but default publish...

Reply:
It's not a problem per-se. There's a couple of reasons why I think it would result in better dev-ex:
* Separate methods communicate more clearly the intention of the caller, without the need of having to read the documentation.
* Did ION does anchoring within the `create` method, so the change would make it more consistent.
* I can't foresee a scenario where a developer wants to create a DHTDid without publishing.

Reply:
> I can't foresee a scenario where a developer wants to create a DHTDid without publishing

It enables private (peer-wise) usage of the DID without needing it to be published.

Reply:
also you may be in an environment where a network call is not possible/desirable